sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -77,59 +83,95 @@ protected IMessage fromByte(byte[] payload) {
     }
 
     @Override
-    protected void setCustomHeaders(MessageHeaders headers, IMessage serviceBusMessage) {
-
-        if (headers.containsKey(MessageHeaders.CONTENT_TYPE)) {
-            Object contentType = headers.get(MessageHeaders.CONTENT_TYPE);
-
-            if (contentType instanceof MimeType) {
-                serviceBusMessage.setContentType(((MimeType) contentType).toString());
-            } else {
-                serviceBusMessage.setContentType((String) contentType);
-            }
-        }
-
-        if (headers.containsKey(MessageHeaders.ID)) {
-            serviceBusMessage.setMessageId(String.valueOf(headers.get(MessageHeaders.ID, UUID.class)));
-        }
-
-        if (headers.containsKey(MessageHeaders.REPLY_CHANNEL)) {
-            serviceBusMessage.setReplyTo(headers.get(MessageHeaders.REPLY_CHANNEL, String.class));
-        }
-
-        if (headers.containsKey(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE)) {
-            Integer integerValue = headers.get(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE, Integer.class);
-            if (null != integerValue) {
-                serviceBusMessage.setScheduledEnqueueTimeUtc(Instant.now().plus(Duration.ofMillis(integerValue)));
-            }
-        }
+    protected void setCustomHeaders(MessageHeaders headers, IMessage message) {
+
+        // Spring MessageHeaders
+        getStringHeader(headers, MessageHeaders.ID).ifPresent(message::setMessageId);
+        getStringHeader(headers, MessageHeaders.CONTENT_TYPE).ifPresent(message::setContentType);
+        getStringHeader(headers, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
+
+        // AzureHeaders.
+        getStringHeader(headers, AzureHeaders.RAW_ID).ifPresent(message::setMessageId);
```

Comment:
I think in the original design, the RAW_ID is used only when servicebusMessage -> springMessage, so do we really need it here?

Reply:
personally I'd like we keep it as it is, to avoid break any user code

Reply:
It's new added feature: Support setting `serviceBusMessage.messageId` by `springMessage`'s header.

And we support 2 ways to set `serviceBusMessage.messageId` by `springMessage` header: `AzureHeaders.RAW_ID` and `ServiceBusMessageHeaders.MESSAGE_ID`, the later one have higher priority.

Reply:
1. rawId from serviceBusMessage -> springMessage
2. rawId from springMessage -> serviceBusMessage

If we only support one of above, our customer may feel weird.